lib/plugins/aws/deploy/compile/events/apiGateway/lib/validate.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
+
+const forEach = require('lodash').forEach;
+const BbPromise = require('bluebird');
+
+module.exports = {
+  validate() {
+    if (!this.serverless.service.resources.Resources) {
+      throw new this.serverless.classes.Error(
+        'This plugin needs access to Resources section of the AWS CloudFormation template');
+    }
+
+    this.options.stage = this.options.stage
+      || (this.serverless.service.defaults && this.serverless.service.defaults.stage)
+      || 'dev';
+    this.options.region = this.options.region
+      || (this.serverless.service.defaults && this.serverless.service.defaults.region)
+      || 'us-east-1';
+
+    // validate stage / region exists in service
+    this.serverless.service.getStage(this.options.stage);
+    this.serverless.service.getRegionInStage(this.options.stage, this.options.region);
+
+    // validate that path and method exists for each http event in service
+    forEach(this.serverless.service.functions, (functionObject) => {
```

Comment:
Thanks. Good catch. Removed the validation in resources.js as it's better to do validation in the first place before proceeding at all.